### PR TITLE
Added completion-at-point-functions for jenkinsfile-mode

### PR DESCRIPTION
**Pull request by CsBigDataHub**
Trying to add completions for Jenkinsfile

### CODE REVIEW

**Comment by CsBigDataHub** [comment]
@john2x friendly ping..

File: jenkinsfile-mode.el [hunk 1]
```diff
@@ -236,10 +236,66 @@ Run this manually when editing this file to get an updated the list of keywords.
 (defvar jenkinsfile-mode-font-lock-defaults
   (append groovy-font-lock-keywords jenkinsfile-mode-font-lock-keywords))
 
+(defun jenkinsfile-pipeline-step-compeletion-at-point ()
```
Comment (john2x) [inline jenkinsfile-mode.el:239]: Can this and the other methods be prefixed with `jenkinsfile-mode-`?

File: jenkinsfile-mode.el [hunk 2]
```diff
@@ -236,10 +236,66 @@ Run this manually when editing this file to get an updated the list of keywords.
...
 ;;;###autoload
 (define-derived-mode jenkinsfile-mode groovy-mode "Jenkinsfile"
   "Major mode for editing Jenkins declarative pipeline files."
-  (setq font-lock-defaults '(jenkinsfile-mode-font-lock-defaults)))
+  (setq font-lock-defaults '(jenkinsfile-mode-font-lock-defaults))
+  (add-hook 'completion-at-point-functions 'jenkinsfile-file-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-directive-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-option-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-pipeline-step-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-core-step-compeletion-at-point nil 'local)
```
Comment (john2x) [inline jenkinsfile-mode.el:297]: Disclaimer: I'm not too familiar with how completion-at-point works. Do these `add-hook` calls need an equivalent `remove-hook` call somewhere else?

**Review by john2x — CHANGES_REQUESTED** [review]
Thanks @CsBigDataHub. A few requests for changes and one question.

**Comment by CsBigDataHub** [comment]
@john2x , added "jenkinsfile-mode" prefix for all the completions.

for info on keyword completion please have a look at http://ergoemacs.org/emacs/elisp_keyword_completion.html

File: jenkinsfile-mode.el (hunk 1 above)
Comment (CsBigDataHub) [reply jenkinsfile-mode.el]: Done.

**Review by john2x — APPROVED** [review]
Thanks @CsBigDataHub

**Pull request merged**